hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/ha/OMFailoverProxyProvider.java

```diff
@@ -318,9 +318,12 @@ synchronized boolean updateLeaderOMNodeId(String newLeaderOMNodeId) {
     if (!currentProxyOMNodeId.equals(newLeaderOMNodeId)) {
       if (omProxies.containsKey(newLeaderOMNodeId)) {
         currentProxyOMNodeId = newLeaderOMNodeId;
+        lastAttemptedOM = currentProxyOMNodeId;
```

Comment:
We are updating lastAttemptedOM to the new OM which will be tried after failover. The assignment should happen before the currentProxyOMNodeId is updated with newLeaderOMNodeId.

Reply:
@hanishakoneru good catch. I added new test to cover this case too. Please take a look at it.